tests/integration/kafka/test_consumer.py

```diff
@@ -77,3 +78,46 @@ def test_get_watermark_offsets_topic_with_one_message(
 
         assert beginning == 0
         assert end == 1
+
+
+def test_assign_after_subscribe_consumes_from_beginning(
```

Comment:
Same comment applies here, this looks useful to pinpoint the issue, but I don't think we should add this test to the code-base.